sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -955,6 +957,18 @@ class Analyzer(
         Project(p.output, newPlan.withNewChildren(newChild :: Nil))
     }
   }
+
+  object CleanAlias extends Rule[LogicalPlan] {
```

Comment:
There are possibly other rules here (`ResolveSortOrders`) that have unnecessary logic if we are going to add this rule.